main_app/igdb_api.py

```diff
@@ -93,34 +100,82 @@ def api_get_last_pages_amount(self) -> int:
         category = 'games/?'
         additional = {
-            'fields': 'name,cover,version_title,rating',
+            'fields': 'name,cover,version_title,rating,genres',
             'limit': self.__limit,
-            'offset': (int(page) - 1) * self.__limit}
-        if not kwargs.get('search'):
-            additional['order'] = 'popularity:desc'
-        filters = self.__generate_filters(**kwargs)
-        additional.update(filters)
+            'offset': (int(page) - 1) * self.__limit
+        }
+        if game_ids:
+            games = ','.join(map(str, game_ids))
+            category = f'games/{games}?'
+        else:
+            if not kwargs.get('search'):
+                additional['order'] = 'popularity:desc'
+            filters = self.__generate_filters(**kwargs)
+            additional.update(filters)
         encoded_url = urlencode(additional, quote_via=quote_plus)
         data = self.__api_get(f'{category}{encoded_url}')
         images_query = list(set(each.get('cover', '') for each in data if each.get('cover')))
-        images = dict(self.api_get_image(images_query, True))
+        images = self.api_get_image(images_query, True)
+        genres_list = []
         for i, each in enumerate(data):
             if each:
                 if each.get('cover'):
                     data[i]['cover'] = images.get(each.get('cover'))
+                if genres:
+                    genres_list.append(data[i].get('genres', [0])[0])
+        if genres:
+            j = 0
+            genre_names = self.api_get_names('genres', genres_list)
+            print(genre_names)
+            for i, each in enumerate(data):
+                if each.get('genres'):
+                    data[i]['genres'] = genre_names[each.get('genres')[0]]
+                    j += 1
         return data
 
     def api_get_image(self, images_id: list, cover=False):
+        images_id = list(set(images_id))
         result = 'covers/' if cover else 'screenshots/'
-        data = self.__api_get(result + ','.join(map(str, images_id)) + '?fields=url')
-        for each in data:
-            yield each.get('id'), 'https:' + each.get('url', '').replace('t_thumb', 't_cover_big' if cover
-                                                         else 't_screenshot_med')
+        data = []
+        if len(images_id) > 10:
+            for i in range(math.floor(len(images_id) / 10)):
+                data.extend(self.__api_get(result + ','.join(map(str, images_id[i * 10:(i + 1) * 10])) + '?fields=url'))
+            data.extend(self.__api_get(result + ','.join(map(str,
+                                                             images_id[math.floor(len(images_id) / 10) * 10:])) +
+                                       '?fields=url'))
+        return {each.get('id'): 'https:' + each.get('url', '').replace('t_thumb', 't_cover_big' if cover
+                                                                         else 't_screenshot_med') for each in data}
```

Comment:
It is hardly readable. Can we replace it with more verbose but clear variant?
```
result = dict()
for each in data:
   if cover:
       replaced_url = each['url'].replace('t_thumb', 't_cover_big') # you can change var name to anywhat you want
   else:
      replaced_url = each['url'].replace('t_thumb', 't_screenshot_med')
   result[each['id']] = f'https:{replaced_url}'
return result
```

If you will return  to such pieces of code later - it will be much more easier to understand logic, eg 5 sec vs 30 sec.
The same to other developers who will work on your code.

In  "zen of python" there's also statement - "Readability counts." 

https://treyhunner.com/2019/03/abusing-and-overusing-list-comprehensions-in-python/